src/main/java/com/github/cameltooling/dap/internal/CamelDebugAdapterServer.java

```diff
@@ -169,6 +170,14 @@ private SetBreakpointsResponse setBreakpointsSync(SetBreakpointsArguments setBre
 		return response;
 	}
 
+	private void addBreakpoint(SourceBreakpoint sourceBreakpoint, String nodeId) {
+		if (sourceBreakpoint.getCondition() != null) {
+			connectionManager.getBacklogDebugger().addConditionalBreakpoint(nodeId, CAMEL_LANGUAGE_SIMPLE, sourceBreakpoint.getCondition());
```

Comment:
Prefer having positive conditions to negative conditions.

It's also better to have

- Preconditions
- Happy path

As it follows natural reading path

```
if (precondition == null) {
   DoSomething
}

DoSometingIfEverythingIsOk

```

Reply:
usually happy path is when a value is not null. I think using `!= null` is very widespread in Java world.

In this case, it is more 2 different paths, than one happy and one handling error or missing information.

So it is:

if (case1) {
   DoSomethingInCase1
} else {
   DoSomethingInCase2
}

This is anyway not code modified in this PR, it is only extracted from existing code to reduce cyclomatic complexity on another function for which the threshold recommended by Sonar has been hit. Created https://issues.redhat.com/browse/FUSETOOLS2-1831then